src/solvers/refinement/string_refinement.cpp

```diff
@@ -1084,21 +1096,49 @@ static exprt negation_of_constraint(const string_constraintt &axiom)
   return negaxiom;
 }
 
+/// Result of the solver `supert` should not be interpreted literally for char
+/// arrays as not all indices are present in the index set.
+/// We populate the values for which the solver has no constraint by padding
+/// the array to the left.
+/// For example an expression `ARRAY_OF(0) WITH [1:=2] WITH [4:=3]` would
+/// be interpreted as `{ 2, 2, 3, 3, 3}`.
+/// \param expr: expression to interpret
+/// \param string_max_length: maximum size of arrays to concider
+/// \return the interpreted expression
+static exprt concretize_array_expression(
+  const exprt &expr, std::size_t string_max_length)
+{
```

Comment:
Actually the input expression should be any expression, we will just replace inside it the arrays that needs to be.
I can update the documentation to make that clearer